From cfe2306e50972107ee06a4bd3a88d553117b67da Mon Sep 17 00:00:00 2001
From: Icarus Sparry <icarus.w.sparry@intel.com>
Date: Tue, 6 Feb 2018 14:54:17 -0800
Subject: [PATCH 09/10] Stateless files might not exist in /etc

With stateless, there might not be a file /etc/group. If useradd is
called with an existing primary group then all of the following will
be true, db->isopen, !db->changed, db->fp==NULL, which causes a crash
when fclose(NULL) is invoked.

Signed-off-by: Icarus Sparry <icarus.w.sparry@intel.com>
---
 lib/commonio.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/lib/commonio.c b/lib/commonio.c
index 31edbaa..9ec7b94 100644
--- a/lib/commonio.c
+++ b/lib/commonio.c
@@ -918,7 +918,10 @@ int commonio_close (struct commonio_db *db)
 	db->isopen = false;
 
 	if (!db->changed || db->readonly) {
-		(void) fclose (db->fp);
+		if (db->fp) {
+			/* File might not exist if stateless */
+			(void) fclose (db->fp);
+		}
 		db->fp = NULL;
 		goto success;
 	}
-- 
2.16.2

